src/Microsoft.Azure.SignalR/HubHost/ServiceConnectionManager.cs

```diff
@@ -29,5 +32,14 @@ public void AddServiceConnection(ServiceConnection serviceConnection)
             var index = StaticRandom.Next(_serviceConnections.Count);
             await _serviceConnections[index].WriteAsync(serviceMessage);
         }
+
+        public async Task WriteAsync(string partitionKey, ServiceMessage serviceMessage)
+        {
+            if (_serviceConnections.Count == 0) return;
+
+            var hash = partitionKey?.GetHashCode() ?? DefaultHashCode;
```

Comment:
if `partitionKey` is `null`, shall we call `WriteAsync(ServiceMessage)` instead? otherwise all the traffic with `null` `partitionKey` goes into one service, is that intentional? Or shall we prevent `null` case?

Reply:
This is just defensive code. Right now the partition key won't be null.

Reply:
Also I don't think we should fall back to the random `WriteAsync`. Cause once this method is called, the intention is to send  message from a fix service connection, right? So it will probably break functionality if a random service connection is used.

Reply:
Then throw instead?

Reply:
Fixed.